### ISSUE TITLE
No enemies on deadlines over 10 days

### ISSUE THREAD

**Issue by DaddyBoard** [Kraykennn/LC-DynamicDeadline#7]
Title says it all really, got a 12 day quota and no enemies spawned until Day 10

Also, why is it we got awarded another day (from 3 days to 4 days) in a run where our quota was 800 and we got 4k worth of loot in the van after the 3 days lol..

**Comment by Kraykennn** [comment]
It depends on if you're using the dynamic method or the  legacy method. In the legacy method, you get extra days based on quota/200 (or a value you set), and the dynamic method is currently based on how much you SELL each quota. (Next update will incorporate ship scrap as well)

The no enemies thing is a bug though, I'll be checking that out once I get back from my shift (January 2nd).

**Comment by ustaalon** [comment]
Hey @Kraykennn I might know the issue, as I also wrote a mod that [deals with deadlines](https://github.com/ustaalon/LethalCompany.ExtraDays) (btw, works alongside with your mod 🙌)

I think the code in the repo here is not up-to-date, right? I'm working on solving this issue also in my mod, as I already found why this is happening. I would like to assist. Feel free to reach out on Discord (username: ustaalon / nickname: Anubis)

**Comment by ustaalon** [comment]
@Kraykennn Maybe you can add my mod as a dependency, as it will fix multiple issues (negative buying rate, enemies spawn issues, and other issues regarding the deadline), https://thunderstore.io/c/lethal-company/p/Anubis/ExtraDaysToDeadline/

What do you think?

**Comment by Kraykennn** [comment]
@ustaalon , Repo code is infact up to date at the moment, got one more update to push then I'm gonna start working on other projects. Adding your mod as a dependency could work, as at the moment I've been too busy to fix these issues myself.  (Except for the negative buying rate, that's something that I don't really want to fix as I want to stop people from selling early.) But yeah, I'll likely do that here soon.

**Comment by ustaalon** [comment]
Sure thing. If you need and help with something let me know 😊

**Comment by Masterkevkev** [comment]
> @Kraykennn Maybe you can add my mod as a dependency, as it will fix multiple issues (negative buying rate, enemies spawn issues, and other issues regarding the deadline), https://thunderstore.io/c/lethal-company/p/Anubis/ExtraDaysToDeadline/
> 
> What do you think?

Upon testing, the deadline was calculated to be 37 days and 0 enemies spawned either inside or outside. Not saying one of the mods is at fault, but it's what occurred in my pack.

**Comment by ustaalon** [comment]
Any error in logs? We can look into it together, if you would like that. My Discord username is ustaalon or here: https://discord.com/channels/1168655651455639582/1190842600534573056

My test case was:
10-15 days of deadline, eclipse moons

**Comment by ustaalon** [comment]
I found the issue, as it's my fault. There is an edge case I thought I got and I accidentally removed it from my mod's lastest release 2.0.4.

I'm working now on fixing this from my end and will update here after the update is published.

**Comment by ustaalon** [comment]
@Kraykennn sorry for the inconvenience, I released a fix for it and now it should be ok. If you need any assistance of testing it together, feel free to drop me a message on Discord

[latest version: 2.0.6](https://thunderstore.io/c/lethal-company/p/Anubis/ExtraDaysToDeadline/)

**Comment by DaddyBoard** [comment]
> and the dynamic method is currently based on how much you SELL each quota. (Next update will incorporate ship scrap as well)

Yeah, we sold 4k worth of loot on the Dynamic mode and got an extra day awarded. It got to the point we had 10k in the ship for a 1,204 quota cos we had like 8 days.
